### PR TITLE
BHBC-819 Add location step functionality

### PR DESCRIPTION
**Pull request by anissa-agahchen**
# Overview

## Links to jira tickets

- {https://quartech.atlassian.net/browse/BHBC-819}

## This PR contains the following changes

 **UI:**
* removed location_decription from project and put it in Location step
* added a picklist of regions

**API:**
* updated the PostProjectObject to receive the location description from the location step
* write the function that sends the projectId and region_name to the project_region table

**Unit tests:**
* updated the API PostProjectObject unit tests to get the location_description from the location object.
* created the API PostProjectRegionObject unit tests.

## This PR contains the following types of changes

- [ ] New feature (change which adds functionality)
- [x] Enhancement (improvements to existing functionality)
- [ ] Bug fix (change which fixes an issue)
- [x] Misc cleanup / Refactoring / Documentation
- [ ] Version change

## Checklist

A list of items that are good to consider when making any changes.

_Note: this list is not exhaustive, and not all items are always applicable._

### General

- [ ] I have performed a self-review of my own code

### Code

- [x] New files/classes/functions have appropriately descriptive names and comment blocks to describe their use/behaviour
- [x] I have avoided duplicating code when possible, moving re-usable pieces into functions
- [ ] I have avoided hard-coding values where possible and moved any re-usable constants to a constants file
- [x] My code is as flat as possible (avoids deeply nested if/else blocks, promise chains, etc)
- [x] My code changes account for null/undefined values and handle errors appropriately
- [x] My code uses types/interfaces to help describe values/parameters/etc, help ensure type safety, and improve readability

### Style

- [x] My code follows the established style conventions
- [ ] My code uses native material-ui components/icons/conventions when possible

### Documentation

- [x] I have commented my code sufficiently, such that an unfamiliar developer could understand my code
- [ ] I have added/updated README's and related documentation, as needed

### Tests

- [x] I have added/updated unit tests for any code I've added/updated
- [ ] I have added/updated the Postman requests/tests to account for any API endpoints I've added/updated

### Linting/Formatting

- [x] I have run the linter and fixed any issues, as needed  
       _See the `lint` commands in package.json_
- [x] I have run the formatter and fixed any issues, as needed  
       _See the `format` commands in package.json_

### SonarCloud

- [ ] I have addressed all SonarCloud Bugs, Vulnerabilities, Security Hotspots, and Code Smells

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes.

## Screenshots

Please add any relevant UI screenshots, if applicable.

### CODE REVIEW

**Comment by NickPhura** [comment]
Looks really good!

The only thing we could add are a few unit tests in `project-queries.test.ts` just to cover the new region function you added.

Updating the postman tests. Specifically the `POST /project (WIP)` endpoint.  All we need to do is add a `location` object to the request body.  This is technically optional, as the region fields aren't required, but its nice for someone else running it so they know what all of the values it supports are, and so if they need to generate a test project, its as filled out as possible.

**Comment by anissa-agahchen** [comment]
Sounds good :-) I'll update the files.

**Comment by NickPhura** [comment]
Also, regarding the 4 code smells.  

The 2 `TODO` smells are false positives in my opinion, as adding TODOs is actually super helpful to keep track of tech debt, etc.  There might be a way to ignore them in the config, but for now I wouldn't worry about them.

And the 2 `Expected an error object to be thrown` smells are just a side effect of typescript I think.  There is likely a way to appease it, but its not going to change the overall functionality, so we can leave it for now.  One of the things I want to do later is to create some helper functions for throwing errors, so its unified across all endpoints.  But not a problem right this second.

File: api/src/paths/project.ts [hunk 1]
```diff
@@ -145,39 +145,44 @@ function createProject(): RequestHandler {
...
-
-            if (!ancillarySpeciesResult || !ancillarySpeciesResult.id) {
-              throw {
-                status: 400,
-                message: 'Failed to insert into ancillary_species table'
-              };
-            }
-          })
-        );
+        if (req.body.species?.ancillary_species && Array.isArray(req.body.species?.ancillary_species)) {
```
Comment (NickPhura) [inline api/src/paths/project.ts:150]: Could you add this style check to the other 2 arrays as well (focal_species, and regions)

Comment (anissa-agahchen) [reply api/src/paths/project.ts:150]: oh right

**Review by NickPhura — APPROVED** [review]
Looks good

**Pull request merged**